omero_marshal/decode/decoders/screen.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# jason@glencoesoftware.com.
+#
+
+from ... import SCHEMA_VERSION
+from .annotation import AnnotatableDecoder
+from omero.model import ScreenI
+
+
+class Screen201501Decoder(AnnotatableDecoder):
+
+    TYPE = 'http://www.openmicroscopy.org/Schemas/OME/2015-01#Screen'
+
+    OMERO_CLASS = ScreenI
+
+    def decode(self, data):
+        v = super(Screen201501Decoder, self).decode(data)
+        self.set_property(v, 'name', data.get('Name'))
+        self.set_property(v, 'description', data.get('Description'))
+        self.set_property(v, 'protocolDescription',
+                          data.get('ProtocolDescription'))
+        self.set_property(v, 'protocolIdentifier',
+                          data.get('ProtocolIdentifier'))
+        self.set_property(v, 'reagentSetDescription',
+                          data.get('ReagentSetDescription'))
+        self.set_property(v, 'reagentSetIdentifier',
+                          data.get('ReagentSetIdentifier'))
+        self.set_property(v, 'type', data.get('Type'))
+        for plate in data.get('Plates', list()):
```

Comment:
If `Plates` is missing you need to skip it. Otherwise, if the collection was unloaded when you encoded it and you are attempting a round trip, for example updating the `name`, you will end up blowing away the current linked Plates.